include/core/mir/geometry/point_generic.h

```diff
@@ -70,24 +70,24 @@ inline constexpr bool operator != (P const& lhs, P const& rhs)
 }
 
 template<typename P, typename std::enable_if<std::is_base_of<detail::PointBase, P>::value, bool>::type = true>
-inline constexpr P operator+(P lhs, typename P::template WrapperType<DeltaXTag> rhs) { return{lhs.x + rhs, lhs.y}; }
+inline constexpr P operator+(P lhs, typename P::template Corresponding<DeltaXTag> rhs) { return{lhs.x + rhs, lhs.y}; }
```

Comment:
This repeated "`typename P::template Corresponding<DeltaXTag>`" incantation is annoying. It could be factored out with something like:

```C++
template<class GeometricType, typename Tag>
using corresponding = typename GeometricType::template Corresponding<Tag>;

inline constexpr P operator+(P lhs, corresponding<P, DeltaXTag> rhs) { return{lhs.x + rhs, lhs.y}; }
```

Reply:
Or even

```C++
template<class GeometricType, typename Tag, typename std::enable_if<std::is_base_of<detail::PointBase, GeometricType>::value, bool>::type = true>
using Convert= typename GeometricType::template Corresponding<Tag>;

template<typename Tag>
using ToCorresponding = Tag;

template<typename P>
inline constexpr P operator+(P lhs, Convert<P, ToCorresponding<DeltaXTag>> rhs) { return{lhs.x + rhs, lhs.y}; }
```